planning/behavior_path_planner/src/planner_manager.cpp

```diff
@@ -341,73 +342,101 @@ std::pair<SceneModulePtr, BehaviorModuleOutput> PlannerManager::runCandidateModu
     return SceneModulePtr();
   }();
 
+  /**
+   * register candidate modules.
+   */
   updateCandidateModules(executable_modules, module_ptr);
 
   return std::make_pair(module_ptr, results.at(module_ptr->name()));
 }
 
 BehaviorModuleOutput PlannerManager::runApprovedModules(const std::shared_ptr<PlannerData> & data)
 {
-  BehaviorModuleOutput output = getReferencePath(data);  // generate root reference path.
+  std::unordered_map<std::string, BehaviorModuleOutput> results;
 
-  bool remove_after_module = false;
+  BehaviorModuleOutput output = getReferencePath(data);
+  results.emplace("root", output);
 
-  for (auto itr = approved_module_ptrs_.begin(); itr != approved_module_ptrs_.end();) {
-    const auto & name = (*itr)->name();
+  /**
+   * execute all approved modules.
+   */
+  for (const auto & module_ptr : approved_module_ptrs_) {
+    output = run(module_ptr, data, output);
+    results.emplace(module_ptr->name(), output);
+  }
```

Comment:
For consistency of using STL, I would suggest either using 
1. `std::for_each`, or
2. `std::transform` with `std::back_inserter(results)`.